### PR TITLE
WIP: Collapsible query and minor UI improvements

### PR DESCRIPTION
**Pull request by ITAYC0HEN**
**Your checklist for this pull request**
- [x] I've read the [contributing guideline](https://github.com/CERT-Polska/mquery/blob/master/CONTRIBUTING.md).
- [x] I've tested my changes by building and running mquery, and testing changed functionality (if applicable)

This PR is a work-in-progress. I applied the prettier for formatting and it looks great on VSCODE, but Github shows huge diff in the code. I'll investigate this tomorrow.

**What is the current behavior?**

- The query pane takes a lot of space when showing the results
- Multiple buttons are too big and steal screen space
- Matched rules are not aligned in results view
- "API Docs" button was stuck in the middle of the screen
- The status label was shown in gray
- Scrollbar contained the progress bar, # of matched samples and status

**What is the new behavior?**

_Click on the GIF to see it in full size_
demo_collapse_query

- Implemented a button to collapse and uncollapse the Query view
  - The collapse button is not visible when no query was executed (e.g writing new query)
  - After clicking the "Query" button, the Query view will collapse since it is less likely to be used
  - When entering an executed query from the /recent page, the Query view will be collapsed
- Implemented a generic function to color both the Status label and the progress bar based on the job's status
- The matched rules labels are now pulled to the right and more spacious
- The buttons in the Query view are smaller now
- The scroll bar is now available only in the Matches table. Hence, the status view with the progress bar 
stays at top
- The "API Docs" button was wrongly aligned. Now it is pulled to the right, assuming the author didn't mean to pull it to the left.

**Test plan**
TODO

**Closing issues**

TODO

### CODE REVIEW

File: src/mqueryfront/src/QueryField.js [hunk 1]
```diff
@@ -52,7 +52,7 @@ class QueryField extends Component {
                     this.props.updateQhash(
                         response.data.query_hash,
                         this.state.rawYara
-                    );
+                    );
```
Comment (msm-code) [inline src/mqueryfront/src/QueryField.js:55]: Nitpick: unnecessary whitespace (this is also the cause of the failed check)

File: src/mqueryfront/src/QueryPage.js [hunk 2]
```diff
@@ -27,6 +28,7 @@ class QueryPage extends Component {
         this.updateQhash = this.updateQhash.bind(this);
         this.updateQueryError = this.updateQueryError.bind(this);
         this.updateQueryPlan = this.updateQueryPlan.bind(this);
+        this.collapsePane = this.collapsePane.bind(this);
```
Comment (msm-code) [inline src/mqueryfront/src/QueryPage.js:31]: shouldn't this be called `togglePane`? (since it's both `collapse` and `uncollapse`)

File: src/mqueryfront/src/App.css [hunk 3]
```diff
@@ -4,10 +4,26 @@
...
-    height: calc(100vh - 95px);
+    height: calc(100vh - 235px);
     overflow-y: scroll;
 }
 
 .mquery-scroll-matches td {
     word-break: break-all;
 }
+
+#query-column {
```
Comment (msm-code) [inline src/mqueryfront/src/App.css:15]: Is it possible to achieve this effect with css classes instead of IDs? AFAIK classes are strongly preferred over IDs in reactjs when possible. I *think* in this case simple class will be enough?

File: src/mqueryfront/src/App.css [hunk 4]
```diff
@@ -4,10 +4,26 @@
...
+    transition: all .3s ease-out;
+}
+
+.is-collapsed {
+    -webkit-transform: translateX(-100%);
+    transform: translateX(-100%);
+    position: absolute;
+}
+
+#results-column {
```
Comment (msm-code) [inline src/mqueryfront/src/App.css:26]: I think it's not used anywhere - can be safely removed

Comment (msm-code) [inline src/mqueryfront/src/App.css:26]: I think it's not used anywhere - can be safely removed

File: src/mqueryfront/src/QueryPage.js [hunk 5]
```diff
@@ -162,16 +170,31 @@ class QueryPage extends Component {
...
+                />
+            </div>
         );
         return (
             <div className="container-fluid">
                 <div className="row">
-                    <div className="col-md-6">
+                    <div
+                        id="query-column"
+                        className={
```
Comment (msm-code) [inline src/mqueryfront/src/QueryPage.js:194]: See comment above, I think this ID can be replaced by class (after the CSS update:
```html
                    <div
                        className={
                            "query-column " + 
                            (this.state.collapsed ? "is-collapsed" : "col-md-5")
                        }
                    >
```

File: src/mqueryfront/src/QueryPage.js [hunk 6]
```diff
@@ -182,7 +205,13 @@ class QueryPage extends Component {
                             updateQueryError={this.updateQueryError}
                         />
                     </div>
-                    <div className="col-md-6" id="status-col">
+                    <div
+                        id="results-query"
```
Comment (msm-code) [inline src/mqueryfront/src/QueryPage.js:209]: Unused ID, can be removed I think

File: src/mqueryfront/src/QueryPage.js [hunk 7]
```diff
@@ -182,7 +205,13 @@ class QueryPage extends Component {
...
+                        className={
+                            this.state.collapsed ? "col-md-12" : "col-md-7"
+                        }
+                        id="status-col"
```
Comment (msm-code) [inline src/mqueryfront/src/QueryPage.js:213]: actually both of them can be removed :thinking:. I don't know why we kept this.

File: src/mqueryfront/src/QueryPage.js (hunk 2 above)
Comment (ITAYC0HEN) [reply src/mqueryfront/src/QueryPage.js:31]: That's tricky. Usually, such elements are called "collapsible" in the Bootstrap word. And the button is toggling it. But the functions themselves are called `collapse()`. It is up to you, whatever you decide :)

Comment (msm-code) [reply src/mqueryfront/src/QueryPage.js:31]: Sounds good, I was making sure I understand the code intention

**Comment by ITAYC0HEN** [comment]
Alright, @msm-code I think it is ready now. All checks are passed.
I removed the animation completely because I couldn't make it work smoothly. I am rusty with boostrap 4 and I think that the mixture of bootsrap3 and 4 in the UI made it more complicated than I think.

So as for now, to collapse is immediate and now animation applied. Thus, no need for element's IDs.
On the same topic, I am not sure I strongly agree with you about not using ID for elements. Unfamiliar with ReactJS best-practices but I don't like creating a class for a single component that will stay single. I am aware fo the debate about using or not using IDs in CSS. But well, no animation - no IDs :)

**Comment by msm-code** [comment]
I see. It looked smooth for me, but I'm the last person you should as so I take your expertise.

Do you have examples of mixed bootstrap3/4 elements? My knowledge of bootstrap is limited to copy&pasting random snippets from the internet until they look good, but I'll happily educate myself and maybe refactor this in a (distant) future.

`I am not sure I strongly agree with you about not using ID for elements` I don't have a *strong* opinion on that topic, just something that I've read about and try to follow. It also makes sense to me as a programmer - ids must be globally unique so eveything with an id is not reusable. Anayway, I wouldn't have a problem with IDs if you insisted, but as you've said - no ids no problem ;)
